diff --git a/fbcode/velox/external/date/date.h b/fbcode/velox/external/date/date.h
--- a/fbcode/velox/external/date/date.h
+++ b/fbcode/velox/external/date/date.h
@@ -86,6 +86,10 @@
 #   pragma warning(disable : 4127)
 #endif
 
+namespace facebook
+{
+namespace velox
+{
 namespace date
 {
 
@@ -7933,7 +7937,9 @@
                  detail::get_units<CharT>(typename Period::type{});
 }
 
-}  // namespace date
+} // namespace date
+} // namespace velox
+} // namespace facebook
 
 #ifdef _MSC_VER
 #   pragma warning(pop)
diff --git a/fbcode/velox/external/date/ios.h b/fbcode/velox/external/date/ios.h
--- a/fbcode/velox/external/date/ios.h
+++ b/fbcode/velox/external/date/ios.h
@@ -32,16 +32,22 @@
 # if TARGET_OS_IPHONE
 #   include <string>
 
+    namespace facebook
+    {
+    namespace velox
+    {
     namespace date
     {
     namespace iOSUtils
     {
-    
+
     std::string get_tzdata_path();
     std::string get_current_timezone();
-    
+
     }  // namespace iOSUtils
     }  // namespace date
+    }  // namespace velox
+    }  // namespace facebook
 
 # endif  // TARGET_OS_IPHONE
 #else   // !__APPLE__
diff --git a/fbcode/velox/external/date/tz.cpp b/fbcode/velox/external/date/tz.cpp
--- a/fbcode/velox/external/date/tz.cpp
+++ b/fbcode/velox/external/date/tz.cpp
@@ -268,6 +268,10 @@
 
 #endif  // !USE_OS_TZDB
 
+namespace facebook
+{
+namespace velox
+{
 namespace date
 {
 // +---------------------+
@@ -3860,6 +3864,8 @@
 }
 
 }  // namespace date
+}  // namespace velox
+}  // namespace facebook
 
 #if defined(__GNUC__) && __GNUC__ < 5
 # pragma GCC diagnostic pop
diff --git a/fbcode/velox/external/date/tz.h b/fbcode/velox/external/date/tz.h
--- a/fbcode/velox/external/date/tz.h
+++ b/fbcode/velox/external/date/tz.h
@@ -143,6 +143,10 @@
 #  endif
 #endif
 
+namespace facebook
+{
+namespace velox
+{
 namespace date
 {
 
@@ -2790,5 +2794,7 @@
 #endif  // !MISSING_LEAP_SECONDS
 
 }  // namespace date
+}  // namespace velox
+}  // namespace facebook
 
 #endif  // TZ_H
diff --git a/fbcode/velox/external/date/tz_private.h b/fbcode/velox/external/date/tz_private.h
--- a/fbcode/velox/external/date/tz_private.h
+++ b/fbcode/velox/external/date/tz_private.h
@@ -34,6 +34,12 @@
 #include <vector>
 #endif
 
+namespace facebook
+{
+
+namespace velox
+{
+
 namespace date
 {
 
@@ -309,6 +315,10 @@
 
 }  // namespace date
 
+}  // namespace velox
+
+}  // namespace facebook
+
 #if defined(_MSC_VER) && (_MSC_VER < 1900)
 #include "tz.h"
 #endif
diff --git a/fbcode/velox/external/date/iso_week.h b/fbcode/velox/external/date/iso_week.h
--- a/fbcode/velox/external/date/iso_week.h
+++ b/fbcode/velox/external/date/iso_week.h
@@ -27,12 +27,6 @@
 
 #include <climits>
 
-namespace facebook
-{
-namespace velox
-{
-namespace date
-{
 namespace iso_week
 {
 
@@ -1763,8 +1757,5 @@
 }
 
 } // namespace iso_week
-} // namespace date
-} // namespace velox
-} // namespace facebook
 
 #endif  // ISO_WEEK_H
