diff --git a/velox/external/date/date.h b/velox/external/date/date.h
index ccbd4587f..ac6d636dd 100644
--- a/velox/external/date/date.h
+++ b/velox/external/date/date.h
@@ -1,5 +1,5 @@
-#ifndef DATE_H
-#define DATE_H
+#ifndef VELOX_DATE_H
+#define VELOX_DATE_H

 // The MIT License (MIT)
 //
@@ -7949,4 +7949,4 @@ operator<<(std::basic_ostream<CharT, Traits>& os,
 # pragma GCC diagnostic pop
 #endif

-#endif  // DATE_H
+#endif  // VELOX_DATE_H
