diff --git a/velox/external/tzdb/zoned_time.h b/velox/external/tzdb/zoned_time.h
--- a/velox/external/tzdb/zoned_time.h
+++ b/velox/external/tzdb/zoned_time.h
@@ -215,4 +215,21 @@
       __lhs.get_sys_time() == __rhs.get_sys_time();
 }
 
+template <class CharT, class Traits, class Duration, class TimeZonePtr>
+std::basic_ostream<CharT, Traits>& to_stream(
+    std::basic_ostream<CharT, Traits>& os,
+    const CharT* fmt,
+    const zoned_time<Duration, TimeZonePtr>& tp) {
+  using duration = typename zoned_time<Duration, TimeZonePtr>::duration;
+  using LT = date::local_time<duration>;
+  auto const st = tp.get_sys_time();
+  auto const info = tp.get_time_zone()->get_info(st);
+  return to_stream(
+      os,
+      fmt,
+      LT{(st + info.offset).time_since_epoch()},
+      &info.abbrev,
+      &info.offset);
+}
+
 } // namespace facebook::velox::tzdb
